lib/plugins/aws/package/compile/events/stream/index.js

```diff
@@ -134,6 +149,17 @@ class AwsCompileStreamEvents {
               if (typeof event.stream.enabled !== 'undefined') {
                 Enabled = event.stream.enabled ? 'True' : 'False';
               }
+              if (
+                event.stream.consumer &&
+                ['boolean', 'string', 'object'].indexOf(typeof event.stream.consumer) < 0
```

Comment:
Let's skip such validation. In JS world, more natural handling would be as:

```javascript
if (event.stream.consumer) {
  // There's a consumer config
  if (event.stream.consumer === true) {
    // 'true' case
  } else {
    // object case
    if (!event.stream.consumer.requiredProperty) {
      throw new ServerlessError("Missing ...");
    }
  }
}
```

We'll also have schema  based validation brought with: https://github.com/serverless/serverless/issues/6562 so incorporating sophisticated type checks into logic now will add unnecessary maintenance burden.

--- 

So summarizing, I think we should not do any validation here, as _object case_ should be validated by AWS directly, while other cased are picked in straightforward way.

Reply:
Oh okay. I don't write JS every day so I appreciate these tips :)

#6562 looks great!